### PR TITLE
SLING-11355: sling discovery bundles: update to sling 48

### PR DESCRIPTION
**Pull request by ashokmca07**

### CODE REVIEW

**Review by stefan-egli — COMMENTED** [review]
Generally lgtm, just added 1 comment that I think might be worth looking at.

Also, not sure if that should be a worry or not: if I compile with java 8 then things work fine, with java 11 I get 
```
[ERROR] org.apache.sling.discovery.oak.TopologyWebConsolePluginTest.testResourceResolverFactory  Time elapsed: 11.354 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at org.apache.jackrabbit.core.query.AQTQueryFactory.getSupportedLanguages(AQTQueryFactory.java:34)
...
Caused by: java.lang.IllegalArgumentException: org.apache.jackrabbit.spi.commons.query.QueryTreeBuilder is not an ImageIO SPI class
	at java.desktop/javax.imageio.spi.ServiceRegistry.checkClassAllowed(ServiceRegistry.java:722)
	at java.desktop/javax.imageio.spi.ServiceRegistry.lookupProviders(ServiceRegistry.java:170)
	at org.apache.jackrabbit.spi.commons.query.QueryTreeBuilderRegistry.<clinit>(QueryTreeBuilderRegistry.java:55)
```

But I think it might be unrelated (will recheck)

**Comment by stefan-egli** [comment]
Sonar also complaining, in particular about missing coverage of [Config|https://sonarcloud.io/component_measures?metric=new_coverage&selected=apache_sling-org-apache-sling-discovery-oak%3Asrc%2Fmain%2Fjava%2Forg%2Fapache%2Fsling%2Fdiscovery%2Foak%2FConfig.java&view=list&pullRequest=9&id=apache_sling-org-apache-sling-discovery-oak] .. perhaps not super critical

**Comment by stefan-egli** [comment]
PS: I've just opened https://github.com/apache/sling-org-apache-sling-discovery-oak/pull/10 - but that one can easily be rebased, so I suggest we merge this one first - depending on timing

File: src/main/java/org/apache/sling/discovery/oak/SynchronizedClocksHealthCheck.java [hunk 1]
```diff
@@ -54,12 +50,12 @@
...
-        @Property(name = HealthCheck.NAME, value = "Synchronized Clocks", description = "Health Check name", label = "Name"),
-        @Property(name = HealthCheck.TAGS, unbounded = PropertyUnbounded.ARRAY, description = "Health Check tags", label = "Tags"),
-        @Property(name = HealthCheck.MBEAN_NAME, value = "slingDiscoveryOakSynchronizedClocks", description = "Health Check MBean name", label = "MBean name") })
-@Service(value = HealthCheck.class)
+@Component(immediate = true, service = HealthCheck.class,
+        property = {
+                HealthCheck.NAME + "=Synchronized Clocks",
+                HealthCheck.MBEAN_NAME + "=slingDiscoveryOakSynchronizedClocks"
+        })
+        //label = "Apache Sling Discovery Oak Synchronized Clocks Health Check")
```
[the remark this answers is not in the document]
Comment (ashokmca07) [reply src/main/java/org/apache/sling/discovery/oak/SynchronizedClocksHealthCheck.java:58]: I have fixed the labels, now config labels are coming ✅

**Comment by ashokmca07** [comment]
> Generally lgtm, just added 1 comment that I think might be worth looking at.
> 
> Also, not sure if that should be a worry or not: if I compile with java 8 then things work fine, with java 11 I get
> 
> ```
> [ERROR] org.apache.sling.discovery.oak.TopologyWebConsolePluginTest.testResourceResolverFactory  Time elapsed: 11.354 s  <<< ERROR!
> java.lang.ExceptionInInitializerError
> 	at org.apache.jackrabbit.core.query.AQTQueryFactory.getSupportedLanguages(AQTQueryFactory.java:34)
> ...
> Caused by: java.lang.IllegalArgumentException: org.apache.jackrabbit.spi.commons.query.QueryTreeBuilder is not an ImageIO SPI class
> 	at java.desktop/javax.imageio.spi.ServiceRegistry.checkClassAllowed(ServiceRegistry.java:722)
> 	at java.desktop/javax.imageio.spi.ServiceRegistry.lookupProviders(ServiceRegistry.java:170)
> 	at org.apache.jackrabbit.spi.commons.query.QueryTreeBuilderRegistry.<clinit>(QueryTreeBuilderRegistry.java:55)
> ```
> 
> But I think it might be unrelated (will recheck)

@stefan-egli you are right, it was breaking because of `org.apache.sling.commons.testing` which is still pointing to old jackrabbit version. but It should work now, could you please give a try again?

**Review by stefan-egli — APPROVED** [review]
+1, lgtm now

**Pull request merged**